src/main/java/seedu/weme/ui/MemeGridPanel.java

```diff
@@ -21,10 +25,14 @@
     @FXML
     private GridView<Meme> memeGridView;
 
-    public MemeGridPanel(ObservableList<Meme> memeList) {
+    private ObservableMap<String, Integer> likeData;
+
+    public MemeGridPanel(ObservableList<Meme> memeList, ObservableMap<String, Integer> likeData) {
         super(FXML);
         memeGridView.setItems(memeList);
-        memeGridView.setCellFactory(gridView -> new MemeGridViewCell());
+        memeGridView.setCellFactory(listView -> new MemeGridViewCell());
+        this.likeData = likeData;
+        likeData.addListener((MapChangeListener<String, Integer>) change -> memeGridView.autosize());
```

Comment:
Yes. It somehow forces the GridView to refresh and currently the most cost effective implementation. Will change as it affects my component's rating as well. On a side note, an ObservableList cannot be bind with an ObservableMap. In 1.3 this will be StatsEngine which might have features like Views or other stuff. It is imperative to change it out.